jdaviz/core/template_mixin.py

```diff
@@ -1019,8 +1019,9 @@ def selected_obj(self):
         # that.  For imviz, this will mean we won't be able to loop through each of the viewers,
         # but the original viewer should have access to all the subsets.
         for viewer_ref in self.viewer_refs:
-            match = self.app.get_subsets_from_viewer(viewer_ref,
-                                                     subset_type=subset_type).get(self.selected)
+            match = self.app.get_subsets(self.selected,
+                                         spatial_only=subset_type == 'spatial',
+                                         spectral_only=subset_type == 'spectral')
```

Comment:
Just curious, why do you look for spatial or spectral only here? Can `self.selected` be multiple subset names?

Reply:
mainly to map the old call to `get_subsets_from_viewer` as directly as possible.  Is it more efficient or less efficient (in `get_subsets`) to request only one type vs not requesting.  If it just adds more overhead, then I think it can safely be removed here.

Reply:
It shouldn't change anything calling it with or without those parameters, assuming `self.selected` is a single subset. I'll leave it up to you if you want to leave it in for continuities sake or remove it.

Reply:
ok, I agree its not really necessary so let's remove a few lines of code!